packages/NodeTypeResolver/NodeTypeResolver.php

```diff
@@ -121,6 +129,17 @@ public function isObjectType(Node $node, ObjectType $requiredObjectType): bool
     }
 
     public function getType(Node $node): Type
+    {
+        $key = $this->getNodeKey($node);
+
+        if (! array_key_exists($key, $this->resolvedTypes)) {
```

Comment:
@staabm this may need to force redefine type when startTokenPos and/or endTokenPos has different with original node (it means node just refactored):

```suggestion
        $originalNode = $node->getAttribute(AttributeKey::ORIGINAL_NODE);
        if (! array_key_exists($key, $this->resolvedTypes) || ! $originalNode instanceof Node || $originalNode->getStartTokenPos() !== $node->getStartTokenPos() || $originalNode->getEndTokenPos() !== $node->getEndTokenPos()) {
```